e2e_correctness/query_neo_mem.py

```diff
@@ -0,0 +1,275 @@
+
+Workaround would be to add API to create nodes by ids on Memgraph when importing via import_util.
+"""
+
+import gqlalchemy
+import json
+import logging
+import neo4j
+import re
+
+from typing import Any, Dict, List
+
+logging.basicConfig(format="%(asctime)-15s [%(levelname)s]: %(message)s")
+logger = logging.getLogger("query_neo_mem")
+logger.setLevel(logging.DEBUG)
+
+
+class Vertex:
+    def __init__(self, id: int, labels: List[str], properties: Dict[str, Any]):
+        self._id = id
+        self._labels = labels
+        self._properties = properties
+        self._labels.sort()
+
+    @property
+    def id(self) -> int:
+        return self._id
+
+    def __str__(self) -> str:
+        return f"Vertex: {self._id}, {self._labels}, {self._properties}"
+
+    def __eq__(self, other):
+        assert isinstance(
+            other, Vertex
+        ), f"Comparing vertex with object of type {type(other)}"
+        logger.debug(f"comparing Vertex with {self._id} to {other._id}")
+        if self._id != other._id:
+            logger.debug(f"_id different: {self._id} vs {other._id}")
+            return False
+        if self._labels != other._labels:
+            logger.debug(
+                f"_labels different between {self._id} and {other._id}: {self._labels} vs {other._labels}"
+            )
+            return False
+        for k, v in self._properties.items():
+            if k not in other._properties:
+                logger.debug(f"Property with key {k} not in {other._properties.keys()}")
+                return False
+            if v != other._properties[k]:
+                logger.debug(f"Value {v} not equal to {other._properties[k]}")
+                return False
+        return True
+
+
+class Edge:
+    def __init__(
+        self, from_vertex: int, to_vertex: int, label: str, properties: Dict[str, Any]
+    ):
+        self._from_vertex = from_vertex
+        self._to_vertex = to_vertex
+        self._label = label
+        self._properties = properties
+
+    @property
+    def from_vertex(self) -> int:
+        return self._from_vertex
+
+    @property
+    def to_vertex(self) -> int:
+        return self._to_vertex
+
+    def __eq__(self, other):
+        assert isinstance(
+            other, Edge
+        ), f"Comparing Edge with object of type: {type(other)}"
+        logger.debug(
+            f"comparing Edge ({self._from_vertex}, {self._to_vertex}) to\
+              ({other._from_vertex, other._to_vertex})"
+        )
+        # Return True if self and other have the same length
+        if self._from_vertex != other._from_vertex:
+            logger.debug(
+                f"From vertex is different {self._from_vertex} <> {other._from_vertex}"
+            )
+            return False
+        if self._to_vertex != other._to_vertex:
+            logger.debug(
+                f"From vertex is different {self._to_vertex} <> {other._to_vertex}"
```

Comment:
This is a to-vertex; also, maybe use "source vertex" and "destination vertex" in the logger/debug messages.